playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/ExoPlayerWrapper.java

```diff
@@ -101,7 +110,50 @@ private void createPlayer() {
                 DefaultLoadControl.DEFAULT_BUFFER_FOR_PLAYBACK_AFTER_REBUFFER_MS);
         loadControl.setBackBuffer((int) TimeUnit.MINUTES.toMillis(5), true);
         trackSelector = new DefaultTrackSelector(context);
-        exoPlayer = new ExoPlayer.Builder(context, new DefaultRenderersFactory(context))
+        final int skipSilenceDurationUs;
+        switch (skipSilence) {
+            case MILD:
+                skipSilenceDurationUs = 300_000;
+                break;
+            case MEDIUM:
+                skipSilenceDurationUs = 250_000;
+                break;
+            case AGGRESSIVE:
+                skipSilenceDurationUs = 150_000;
+                break;
+            default:
+                skipSilenceDurationUs = 0;
+                break;
+        }
+        exoPlayer = new ExoPlayer.Builder(context, new DefaultRenderersFactory(context) {
+            @Override
+            protected AudioSink buildAudioSink(
+                    Context context,
+                    boolean enableFloatOutput,
+                    boolean enableAudioTrackPlaybackParams,
+                    boolean enableOffload) {
+                // https://github.com/google/ExoPlayer/blob/r2.14.2/library/core/src/main/java/com/google/android/exoplayer2/DefaultRenderersFactory.java#L637
+                return new DefaultAudioSink.Builder()
+                        .setAudioCapabilities(AudioCapabilities.getCapabilities(context))
+                        .setAudioProcessorChain(
+                                new DefaultAudioProcessorChain(
+                                        new SilenceSkippingAudioProcessor(
+                                                skipSilenceDurationUs,
+                                                skipSilenceDurationUs,
```

Comment:
Padding and duration serve different purposes. I think they should be treated differently then.